src/ol/style/regularshape.js

```diff
@@ -401,11 +431,12 @@ ol.style.RegularShape.prototype.draw_ = function(renderOptions, context, x, y) {
   context.translate(x, y);
 
   context.beginPath();
+  var points = this.points_;
   if (this.radius2_ !== this.radius_) {
-    this.points_ = 2 * this.points_;
+    points = 2 * points;
```

Comment:
Still, I would like to keep the amount of both
* code in this PR, and
* the number of changes for users
minimal.

If I construct a regular polygon with two circles each having 5 points, the total number of points is 10, right? So the current behaviour makes some sense (to me at least).

For this PR I suggest to not change the semantics of that method. Feel free to open an issue where to discuss this. I would not make that change in this PR as it makes it more complicated than it has to be.

It is basically the same you did with the `Style` test file. I would have left it where it is, to keep the changes to review for this PR minimal. (It is OK that you did move it, no worries to change that back again.). An issue where that is discussed would have been enough, I guess.

Don't get me wrong, I **really** appreciate the time and effort you have put into this already. I honestly want to get this in rather sooner than later, but the less changes you introduce, the less controversial it is.